tests/src/rules/no-useless-path-segments.js

```diff
@@ -0,0 +1,50 @@
+import { test } from '../utils'
+import { RuleTester } from 'eslint'
+
+const ruleTester = new RuleTester()
+const rule = require('rules/no-useless-path-segments')
+
+function runResolverTests(resolver) {
+  ruleTester.run(`no-useless-path-segments (${resolver})`, rule, {
+    valid: [
+      test({ code: 'import "./malformed.js"' }),
+      test({ code: 'import "./test-module"' }),
+      test({ code: 'import "."' }),
+      test({ code: 'import ".."' }),
+      test({ code: 'import fs from "fs"' }),
+    ],
+
+    invalid: [
+      test({
+        code: 'import "./../files/malformed.js"',
+        errors: [ 'Useless path segments for "./../files/malformed.js", should be "../files/malformed.js"'],
+      }),
+      test({
+        code: 'import "./../files/malformed"',
+        errors: [ 'Useless path segments for "./../files/malformed", should be "../files/malformed"'],
+      }),
+      test({
+        code: 'import "../files/malformed.js"',
+        errors: [ 'Useless path segments for "../files/malformed.js", should be "./malformed.js"'],
+      }),
+      test({
+        code: 'import "../files/malformed"',
+        errors: [ 'Useless path segments for "../files/malformed", should be "./malformed"'],
+      }),
+      test({
+        code: 'import "./test-module/"',
```

Comment:
yes, that should handle whether there's an extension or not - i'm saying that "whether this rule can warn on a path or not" actually depends on what's on disk, because the trailing slash is meaningful.